common/src/log.rs

```diff
@@ -1,123 +1,172 @@
-#[cfg(feature = "display")]
-use std::fmt::Write;
-
 use chrono::{DateTime, Utc};
 #[cfg(feature = "display")]
-use crossterm::style::{StyledContent, Stylize};
+use crossterm::style::Stylize;
 use serde::{Deserialize, Serialize};
+use strum::EnumString;
+use tracing::{field::Visit, span, warn, Metadata, Subscriber};
+use tracing_subscriber::Layer;
 #[cfg(feature = "openapi")]
 use utoipa::ToSchema;
 use uuid::Uuid;
 
-use crate::deployment::State;
+/// Used to determine settings based on which backend crate does what
+#[derive(Clone, Debug, EnumString, Eq, PartialEq, Deserialize, Serialize)]
+#[cfg_attr(feature = "display", derive(strum::Display))]
+#[cfg_attr(feature = "openapi", derive(ToSchema))]
+pub enum Backend {
```

Comment:
Maybe `BackendSource` or `BackendService`?

Reply:
Want to keep it pure to what it actually represents, so it can be reused. `Source` scopes it to logger's context, and `Service` is a word we should be more reserved with (everything is a service => confusion). Should move it to a different file then...